packages/core/src/components/list-button/list-button.scss

```diff
@@ -47,18 +167,113 @@
 
 .dso-list-button {
   @include button.base($modifiers: false);
-  @include list-button.root();
+
+  @include _state-styling(colors.$wit, colors.$grijs-20, colors.$grijs-90);
+
+  border-radius: button.$border-radius;
+  border-width: 1px;
+  border-style: solid;
+  font-weight: 600;
+  padding-block: #{units.$u2 - 1};
+  padding-inline: #{units.$u2 - 1} #{units.$u6 - 1};
+  position: relative;
+  text-align: start;
+  white-space: normal;
+  inline-size: 100%;
+  cursor: pointer;
+
+  &:hover,
+  &:focus {
+    @include _state-styling(colors.$wit, button.$secondary-hover-background-color, colors.$grijs-90);
+  }
+
+  &:active,
+  &.active {
+    &,
+    &:hover {
+      @include _state-styling(colors.$wit, button.$secondary-hover-background-color, colors.$grijs-90);
+    }
+  }
+
+  &.disabled,
+  &[disabled] {
+    &,
+    &:hover {
+      @include _state-styling(colors.$wit, colors.$grijs-20, colors.$grijs-60);
+
+      // stylelint-disable-next-line no-duplicate-selectors -- needed to fix sass deprecation warnings: #2724
+      & {
+        opacity: 1;
+      }
+    }
+  }
+
+  &.dso-selected {
+    border-color: colors.$grasgroen;
+    border-width: 1px;
+    box-shadow:
+      inset 0px 0px 0px 1px colors.$grasgroen,
+      inset -1px 0px 0px 0px colors.$grasgroen;
+
+    &.dso-single-count {
+      @media screen and (min-width: media-query-breakpoints.$screen-sm-min) {
+        border-inline-end: 0;
+        box-shadow:
+          colors.$grasgroen 0px 1px 0px 0px inset,
+          colors.$grasgroen 1px 0px 0px 0px inset,
+          colors.$grasgroen 0px -1px 0px 0px inset;
+      }
+    }
+  }
+
+  label {
+    cursor: inherit;
+  }
+
+  > span {
+    display: block;
+  }
 
   .dso-sublabel {
+    font-weight: 400;
+    inline-size: 100%;
     padding-inline-start: selectable.$size + units.$u1;
   }
+
+  dso-icon,
+  svg.di,
```

Comment:
Deze selector verwijst naar de HTML/CSS implementatie van Icon, en die wordt niet in de Web Component implementatie van List Button gebruikt. Daar gebruiken we ook de Web Component implementatie (`<dso-icon>`).

Heb je gecheckt op overbodige SCSS ten opzichte van de JSX? Ik kan mij voorstellen dat de List Button en Input Number meer styling genereerde dan in dit geval nodig is.

Reply:
Weggehaald